### ISSUE TITLE
EPIC [Monitoring V2] RFE: Windows Node Metrics

### ISSUE THREAD

**Issue by aiyengar2** [rancher/rancher#28327]
This issue is an Epic that tracks several other issues around supporting Windows Node Metrics for Monitoring V2.

### Goal

Monitoring V2 should have the ability to deploy an exporter to scrape Windows nodes like Monitoring V1 [`exporter-node-windows`](https://github.com/rancher/system-charts/tree/dev-v2.5/charts/rancher-monitoring/v0.1.3/charts/exporter-node-windows). 

The current Monitoring V1 version is outdated as it still uses the old `wmi-exporter`, so additional changes may be required to the chart in order to support the latest [prometheus-community/windows_exporter](https://github.com/prometheus-community/windows_exporter).

### Tasks

- [x] Upgrade wins and fix build issues (https://github.com/rancher/wins/issues/21)
- [x] Add support for publishing ports (https://github.com/rancher/wins/issues/13)
- [x] Create new Github / Docker repo for windows-exporter_package (https://github.com/rancher/rancher/issues/31148)
- [x] Set up Drone for windows-exporter_package
- [x] Ensure Monitoring components are only deployed on Linux hosts (https://github.com/rancher/rancher/issues/31498)
- [x] Create rancher-windows-exporter and add to Monitoring V2 chart (https://github.com/rancher/rancher/issues/31497)
- [x] Upgrade Windows Cluster bootstrapping process to use new wins (https://github.com/rancher/rancher/issues/31499)

### UI Tasks
- [ ] Add ability to detect or select RKE Windows Cluster Type and set default values https://github.com/rancher/dashboard/issues/2537

### Note

We cannot deploy Monitoring V2 on Windows clusters that were provisioned before https://github.com/rancher/rancher/issues/31499 is merged unless they use the wins upgrade tool tracked in https://github.com/rancher/wins/issues/20.

**Comment by aiyengar2** [comment]
As discussed offline, this issue is currently blocked on an RFE at https://github.com/rancher/wins/issues/13.

TLDR for discussion:

In Monitoring V1, Windows support was enabled via the Monitoring V1 chart alongside a special controller ([ExporterEndpointController](https://github.com/rancher/rancher/blob/master/pkg/controllers/managementagent/monitoring/controlplane_endpoint_sync.go#L15), packaged in Rancher itself) that would inject Windows node IP addresses into a Endpoints resource tied to a Service. This provided a Service that Prometheus could target via a ServiceMonitor, which would then scrape http://<node-ip>:<windows-metrics-port e.g. 9796>/metrics.

However, in Monitoring V2 this controller should not be required. As a result, since there is no clean way to create a Service / Endpoints resource pointing to specific node IP addresses (without using hostNetwork, which is not possible in Windows Pods).

To circumvent this, wins needs to be able to support publishing a port so that the wmi-exporter metrics can be scraped directly from http://<pod-ip>:<windows-metrics-port e.g. 9796>/metrics. This also prevent users from having to keep node port 9796 accessible / open for Monitoring to be able to scrape metrics.

**Comment by thehejik** [comment]
Found two new issues not really related to windows but I guess it'd be nice to resolve them:
* https://github.com/rancher/charts/issues/1037 rancher-monitoring v2 chart is using deprecated APIs
* https://github.com/rancher/charts/issues/1038 Removing rancher-monitoring app leaves cattle-monitoring-system namespace in place

**Comment by aiyengar2** [comment]
> Found two new issues not really related to windows but I guess it'd be nice to resolve them:
> 
> * [rancher/charts#1037](https://github.com/rancher/charts/issues/1037) rancher-monitoring v2 chart is using deprecated APIs
> * [rancher/charts#1038](https://github.com/rancher/charts/issues/1038) Removing rancher-monitoring app leaves cattle-monitoring-system namespace in place

This first issue will be resolved by rebasing to the new upstream, prioritized in https://github.com/rancher/rancher/issues/29871.

The second issue does not need to be resolved; the cattle-monitoring-system namespace being left behind is the correct behavior for two reasons:
1) We always leave behind the Alertmanager Secret in the namespace on an uninstall to prevent us from nuking the Alertmanager configuration on an uninstall / reinstall
2) The namespace that Helm is deployed into is not a Helm managed resource in Helm 3; therefore, the Monitoring chart is not responsible for deleting the namespace it was installed into.
